src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -480,35 +174,66 @@ public async Task InitializeFromProfile(DacDeployOptions options, string profile
             SetOptions(options);
         }
 
+        /// <summary>
+        /// Populates OptionsMapTable with the boolean type properties
+        /// </summary>
+        /// <param name="options"></param>
+        public void PopulateOptionsMapTable(DacDeployOptions options)
+        {
+            // To fill the options map table directly from the boolean type DacDeployoptions
+            PropertyInfo[] dacDeploymentOptionsProperties = options.GetType().GetProperties();
+            foreach (var prop in dacDeploymentOptionsProperties)
+            {
+                if (prop.PropertyType == typeof(System.Boolean))
+                {
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    var displayNameAttribute = prop.GetCustomAttributes<DisplayNameAttribute>().FirstOrDefault();
+                    this.OptionsMapTable[displayNameAttribute.DisplayName] = (DeploymentOptionProperty<bool>)setProp;
```

Comment:
Why key off of the display name (which gets localized) rather than the property name (which doesn't)?

Reply:
So these displaNames as Keys() are using as labels to dsiplay on ADS options tab. If we set them as one of the properties in value, it would be costlier to get all the names to from a label array object. and on each checkbox onchange, we need to again loop all the values.displayName of optionsMapTable to fetch the right key to update the value.value. 

Using property as key wouldn't change the previous functionality of needing a map table again there for [key: "display name"; value:propertyname] to update the optionsMapTable.

Reply:
how do you guarantee that none of the keys won't translate to same value in other languages causing duplicate key error? have you tested it for all languages? using localized value as hash key is risky

Reply:
Converted to key:property and value has displayName, will test the funcitonality in Spanish machine that i got from Matteo to make sure the localization works as expected. Thanks